wordpress-mobile/WordPress-Android
# Issue/13328 restore warning

Parent #13328

This PR implements the "warning" wizard step/view for the restore request.

The basic architecture is a clone of BackupDownloadDetailsViewModel with regards to making the request. What differs is the handling of the onBackPressed and error events. I'm on the fence as to whether this should be an actual wizard step or just a different view in the Details step. I am going to leave it as it for this PR, as I can adjust how the errors are handled separately. 

**Merge Instructions**
- Make sure #13773 has been merged 
- Remove Not Ready For Merge Label 
- Merge as normal

**Cloned Classes**
RestoreWarningAdapter
RestoreWarningFragment
RestoreRequestState
RestoreWarningStateListItemBuilder
restore_warning_fragment.xml

**Notes:**
- Ignore styling - this is being handled separately
- Ignore unit test placement in packages - this is being handled separately
- RestoreActivity & RestoreViewModel (+unit tests) will continue to get updated as I move throughout the steps of the wizard

**To test:**
**Warning**: The Restore site button is hooked up, so if you press it your site will be backed up!
- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag on
- Navigate to ActivityLog
- Tap the more menu
- Select the restore option
- Tap the Restore Site button on the Details View
- Verify 
- (1) The warning view is shown
- (2) Tapping on X or back button returns you to the details view
- (3) Tapping the restore site button will kick off a restore. You'll see a snackbar message, but the next page won't show because it hasn't been implemented

PR submission checklist:
